src/functions/bitcoin/bitcoin-functions.ts

```diff
@@ -58,6 +61,101 @@ export function removeDustOutputs(
   }
 }
 
+export function getDerivedUnspendablePublicKeyCommittedToUUID(
+  vaultUUID: string,
+  bitcoinNetwork: Network
+): Buffer {
+  return deriveUnhardenedPublicKey(
+    getUnspendableKeyCommittedToUUID(vaultUUID, bitcoinNetwork),
+    bitcoinNetwork
+  );
+}
+
+/**
+ * This function retrieves the Bitcoin address used to fund a Vault by analyzing the inputs and outputs of the Funding Transaction.
+ *
+ * @param vault - The Vault object containing the Funding Transaction ID and the User's Public Key.
+ * @param feeRecipient - The Fee Recipient's Public Key or Address.
+ * @param extendedAttestorGroupPublicKey - The Extended Public Key of the Attestor Group.
+ * @param bitcoinNetwork - The Bitcoin Network to use.
+ * @param bitcoinBlockchainAPIURL - The Bitcoin Blockchain URL used to fetch the Funding Transaction.
+ * @returns A promise that resolves to the Funding Bitcoin address.
+ * @throws An error if the Vault Funding Address cannot be determined.
+ */
+export async function getVaultFundingBitcoinAddress(
+  vault: RawVault,
+  feeRecipient: string,
+  extendedAttestorGroupPublicKey: string,
+  bitcoinNetwork: Network,
+  bitcoinBlockchainAPIURL: string
+): Promise<string> {
+  const fundingTransaction = await fetchBitcoinTransaction(
+    vault.fundingTxId,
+    bitcoinBlockchainAPIURL
+  );
+
+  const multisigAddress = createTaprootMultisigPayment(
+    getDerivedUnspendablePublicKeyCommittedToUUID(vault.uuid, bitcoinNetwork),
+    deriveUnhardenedPublicKey(extendedAttestorGroupPublicKey, bitcoinNetwork),
+    Buffer.from(vault.taprootPubKey, 'hex'),
+    bitcoinNetwork
+  ).address;
+
+  const feeRecipientAddress = getFeeRecipientAddress(feeRecipient, bitcoinNetwork);
+
+  const inputAddresses = uniq(
+    fundingTransaction.vin.map(input => input.prevout.scriptpubkey_address)
+  );
+
+  const isMultiSigAddress = (address: string) => address === multisigAddress;
+
+  // If the only input is the MultiSig address, it is a withdrawal transaction.
+  // Therefore, the funding address is the non-fee recipient output address.
+  // If there is a single non-MultiSig input that is not the MultiSig address or multiple inputs, it is a funding/deposit transaction.
```

Comment:
This is not worded well, can change to this:

"If there is a single non-MultiSig input that is not from the MultiSig address, or if there are multiple inputs, it is a funding/deposit transaction."